<?xml version="1.0" ?>
<launch>
  <arg name="launch_rviz" default="true" />
  <arg name="semantic_grasping" default="false" />

  <node name="detect_grasps" pkg="gpd_ros" type="detect_grasps" output="screen">
    <param name="cloud_type" value="0" />
    <param name="cloud_topic" value="/cloud_stitched" />
    <param name="samples_topic" value="" />
    <param name="config_file" value="$(find grasp_demo)/config/gpd.cfg" />
    <param name="rviz_topic" value="plot_grasps" />
  </node>

  <!-- <include file="$(find grasp_demo)/launch/voxbloxpp.launch"/> -->

  <!-- <node pkg="grasp_demo" type="simple_scan.py" name="scan_action_node" output="screen"/> -->
  <node pkg="grasp_demo" type="voxbloxpp_scan.py" name="scan_action_node" output="screen"/>
  <node pkg="grasp_demo" type="grasp_execution.py" name="grasp_execution_action_node" output="screen"/>
  <node pkg="grasp_demo" type="drop_node.py" name="drop_action_node" output="screen"/>

  <group if="semantic_grasping">
    <node pkg="grasp_demo" type="semantic_grasp_selection.py" name="semantic_grasp_selection_node" output="screen" />
  </group>
  <group unless="semantic_grasping">
    <node pkg="grasp_demo" type="grasp_selection.py" name="grasp_selection_action_node" output="screen"/>
  </group>

  <node pkg="grasp_demo" type="run_bt.py" name="behaviour_tree_node" output="screen"/>
  <node pkg="rqt_py_trees" type="rqt_py_trees" name="rqt_viewer"/>

  <node if="$(arg launch_rviz)" type="rviz" name="rviz" pkg="rviz" args="-d $(find grasp_demo)/config/rviz.rviz"/>

</launch>
